### PR TITLE
Added fletching table block

### PR DESCRIPTION
**Pull request by ImDaRealAni**
Addition of the fletching table block.

This time, without a bunch of other files. Once again, I forgot about my warning to avoid using "git add ."

Documentation included.

### CODE REVIEW

File: server/block/fletching_table.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	return newFuelInfo(time.Second * 15)
+}
+
+// EncodeItem ...
+func (FletchingTable) EncodeItem() (name string, meta int16) {
+	return "minecraft:fletching_table", 0
+}
+
+// EncodeBlock ...
+func (FletchingTable) EncodeBlock() (name string, properties map[string]any) {
```
Comment (abimek) [inline server/block/fletching_table.go:35]: For the return types you don't need to assign them a name, you can just declare them as (string, map[string]any). Same goes for EncodeItem().

Comment (ImDaRealAni) [reply server/block/fletching_table.go]: Thanks, Scarce!

File: server/block/fletching_table.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"time"
+)
+
+// FletchingTable is a block in villages that turn an unemployed villager into a Fletcher.
+type FletchingTable struct {
+	solid
+	bassDrum
```
Comment (DaPigGuy) [inline server/block/fletching_table.go:11]:
```suggestion
	bass
```

File: server/block/fletching_table.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+// BreakInfo ...
+func (f FletchingTable) BreakInfo() BreakInfo {
+	return newBreakInfo(2.5, alwaysHarvestable, axeEffective, silkTouchOnlyDrop(f))
```
Comment (DaPigGuy) [inline server/block/fletching_table.go:16]:
```suggestion
	return newBreakInfo(2.5, alwaysHarvestable, axeEffective, oneOf(f))
```

Comment (ImDaRealAni) [reply server/block/fletching_table.go]: Thanks again, Pig!

**Review by Sandertv — APPROVED** [review]
This looks good to me! Thanks for the PR!

**Pull request merged**